runtime/compiler/z/codegen/S390PrivateLinkage.cpp

```diff
@@ -1649,6 +1659,140 @@ J9::Z::PrivateLinkage::buildNoPatchingVirtualDispatchWithResolve(TR::Node *callN
    return cursor;
    }
 
+TR::Instruction *
+J9::Z::PrivateLinkage::buildNoPatchingIPIC(TR::Node *callNode, TR::RegisterDependencyConditions *dependencies, TR::Register *vftReg, intptr_t virtualThunk)
+   {
+   /**
+    * TODO: Right now this function generates all the instruction as we are
+    * fixing the number of PIC slots whereas we can change number of slots at compile
+    * time for regular JIT compiled code. It is not that complicated to do that for
+    * readOnlyCode.
+    * We can create a different data structure to hold the J9Class and method address
+    * size of which can be fixed at compilation time and use that.
+    */
+   OMR::CCData *codeCacheData = cg()->getCodeCache()->manager()->getCodeCacheData();
+   OMR::CCData::index_t index;
+
+   if (!(codeCacheData->put(NULL, sizeof(ccInterfaceData), 16, NULL, index)))
+      {
+      cg()->comp()->failCompilation<TR::CompilationException>("Could not allocate interface dispatch metadata");
+      }
+   ccInterfaceData *ccInterfaceDataAddress = codeCacheData->get<ccInterfaceData>(index);
+
+   // Intialization of CCData
+   ccInterfaceDataAddress->slot1Class  = 0;
+   ccInterfaceDataAddress->slot1Method = 0;
+   ccInterfaceDataAddress->slot2Class  = 0;
+   ccInterfaceDataAddress->slot2Method = 0;
+   ccInterfaceDataAddress->slot3Class  = 0;
+   ccInterfaceDataAddress->slot3Method = 0;
+
+   ccInterfaceDataAddress->cpAddress = reinterpret_cast<intptr_t>(callNode->getSymbolReference()->getOwningMethod(cg()->comp())->constantPool());
+   ccInterfaceDataAddress->cpIndex = (intptr_t) callNode->getSymbolReference()->getCPIndexForVM();
```

Comment:
`static_cast` needed here.

Reply:
Fixed in https://github.com/eclipse/openj9/compare/ab82c5d254e0315b3c65722371fda75cc1d79a10..3282a01df297676d33999fca192d2613b5db1b09

Reply:
Fixed in https://github.com/eclipse/openj9/compare/ab82c5d254e0315b3c65722371fda75cc1d79a10..3282a01df297676d33999fca192d2613b5db1b09